dds_cli/data_putter.py

```diff
@@ -169,6 +170,76 @@ def put(
                         for x in [y.id for y in progress.tasks if y.fields.get("step") != "put"]
                     ]
 
+        # LOG.warning(f"test 1: {putter.filehandler.failed}")
+        if pathlib.Path(putter.failed_delivery_log).is_file():
+            LOG.warning(
+                "Some uploaded files could not be added to the database %s.\n",
+                str(putter.failed_delivery_log),
+            )
+            LOG.warning("Attempting to add them to the database.")
+
+            path_to_file = str(putter.failed_delivery_log)
+            with open(path_to_file, "r", encoding="utf-8") as json_f:
+                failed = json.load(json_f)
```

Comment:
No need to convert to string
```suggestion
            with putter.failed_delivery_log.open(mode= "r", encoding="utf-8") as json_f:
                failed = json.load(json_f)
```